src/NFTDriver.sol

```diff
@@ -30,8 +30,16 @@ contract NFTDriver is ERC721Burnable, ERC2771Context, Managed {
     DripsHub public immutable dripsHub;
     /// @notice The driver ID which this driver uses when calling DripsHub.
     uint32 public immutable driverId;
-    /// @notice The ERC-1967 storage slot holding a single `uint256` counter of minted tokens.
-    bytes32 private immutable _mintedTokensSlot = _erc1967Slot("eip1967.nftDriver.storage");
+    /// @notice The ERC-1967 storage slot holding a single `NFTDriverStorage` structure.
+    bytes32 private immutable _nftDriverStorageSlot = _erc1967Slot("eip1967.nftDriver.storage");
+
+    struct NFTDriverStorage {
+        /// @notice The number of tokens minted without salt.
+        uint64 mintedTokens;
+        /// @notice The salts already used for minting tokens.
+        /// The keys are the minter address and the salt, the value is true if the salt is used.
+        mapping(address => mapping(uint64 => bool)) isSaltUsed;
```

Comment:
Why is this needed? We could just use the ERC721 `_exists` function and check if a token exists.

Reply:
We should think about the behaviour we would like to see together with `burn`.

Reply:
What should happen after a token burn. Should it be possible to mint it again? The current design wouldn't allow that.

Reply:
Yes, I just realized otherwise it would result in unexpected behavior. 
For example, the `minter` could have transferred the token to Alice. Alice wants to burn the token some funds regardless that some funds are still waiting for here.

In that case,  it would be not okay if the `minter` can just `mint` the same `tokenId` again and access the funds.

I see two options:
a) the additional mapping for the `salt` which is not reset in the `burn` case. which effectively prevents the re-mint
b) remove the ability to `burn` tokens at all and safe some gas

Reply:
This PR goes with the `a)` option. IMO `b)` is a bit heavy handed, official burning is a great way to officially signal abandonment and freezing of a user ID.